### PR TITLE
build: Simplify builds

### PR DESCRIPTION
**Pull request by alexec**
Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [x] I've signed the CLA.
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My builds are green. Try syncing with master if they are not. 
* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).

@simster7 @sarabala1979 I've been spending too much time fixing builds recently and want to simplify them.

This change will mean that we no longer build branch images (i.e. no images name `argoproj/argocli:my-branch` instead, they will be latest. You must run `make clean` whenever you change branch to get new images.

### CODE REVIEW

**Comment by alexec** [comment]
I've re-enabled up-to-date with master check.

File: .github/workflows/ci-build.yaml [hunk 1]
```diff
@@ -72,9 +68,10 @@ jobs:
           echo '127.0.0.1 minio'    | sudo tee -a /etc/hosts
           echo '127.0.0.1 postgres' | sudo tee -a /etc/hosts
           echo '127.0.0.1 mysql'    | sudo tee -a /etc/hosts
-          mkdir dist
+          mkdir -p dist
           mkdir -p /tmp/log/argo-e2e
-          KUBECONFIG=~/.kube/config make start PROFILE=$PROFILE E2E_EXECUTOR=docker DEV_IMAGE=true > /tmp/log/argo-e2e/argo.log &
+          git fetch --tags
```
Comment (alexec) [inline .github/workflows/ci-build.yaml:73]: this was a key fix

File: .github/workflows/ci-build.yaml [hunk 2]
```diff
@@ -72,9 +68,10 @@ jobs:
...
+          KUBECONFIG=~/.kube/config make start PROFILE=$PROFILE E2E_EXECUTOR=docker DEV_IMAGE=true 2>&1 > /tmp/log/argo-e2e/argo.log &
```
Comment (alexec) [inline .github/workflows/ci-build.yaml:74]: also pipe stderr

File: .github/workflows/ci-build.yaml [hunk 3]
```diff
@@ -102,14 +99,10 @@ jobs:
   codegen:
     name: Codegen
     runs-on: ubuntu-latest
-    timeout-minutes: 10
+    timeout-minutes: 15
```
Comment (alexec) [inline .github/workflows/ci-build.yaml:102]: occasionally this takes longer than 10m

File: .github/workflows/ci-build.yaml [hunk 4]
```diff
@@ -21,10 +21,6 @@ jobs:
     steps:
       - name: Checkout code
         uses: actions/checkout@v2
-        with:
```
Comment (alexec) [inline .github/workflows/ci-build.yaml:24]: @simster7 thoughts on this change?

File: docs/resuming-workflow-via-automation.md [hunk 5]
```diff
@@ -1,4 +1,4 @@
-# Resume The Template
+# Resume A Workflow
```
Comment (alexec) [inline docs/resuming-workflow-via-automation.md:1]: error

File: docs/cli/argo_list.md [hunk 6]
```diff
@@ -22,6 +22,7 @@ argo list [flags]
       --older string            List completed workflows finished before the specified duration (e.g. 10m, 3h, 1d)
   -o, --output string           Output format. One of: wide|name
       --prefix string           Filter workflows by prefix
+      --resubmitted             Show only resubmitted workflows
```
Comment (alexec) [inline docs/cli/argo_list.md:25]: did not run `make docs`

File: Makefile [hunk 7]
```diff
@@ -6,7 +6,7 @@ OUTPUT_IMAGE_ARCH ?= amd64
 BUILD_DATE             = $(shell date -u +'%Y-%m-%dT%H:%M:%SZ')
 GIT_COMMIT             = $(shell git rev-parse HEAD)
 GIT_REMOTE             = origin
-GIT_BRANCH             = $(shell git rev-parse --abbrev-ref=loose HEAD | sed 's/heads\///')
+GIT_BRANCH             = $(shell git rev-parse --abbrev-ref=loose HEAD | sed 's/heads\///' | sed 's/pull\///')
```
Comment (alexec) [inline Makefile:9]: this is a sha hash in PRs

File: .github/workflows/ci-build.yaml (hunk 4 above)
Comment (simster7) [reply .github/workflows/ci-build.yaml:24]: All good with me as long as it plays nice with `GIT_BRANCH             = $(shell git branch --show-current)`, which was the original reason for the code being this way

**Pull request merged**